pkg/core/template/template_test.go

```diff
@@ -19,5 +19,87 @@ func TestTemplateRender(t *testing.T) {
 			_, err = ParseTextTemplate("test", template, nil)
 			So(err, ShouldBeError, "UnexpectedError: rendered template is too large")
 		})
+		Convey("should auto-escape templates", func() {
+			template := `
+			<!DOCTYPE html>
+			<html lang="en">
+			
+			<head>
+				<meta charset="utf-8" />
+				<title>My App</title>
+				<style type="text/css">
+					body {
+						background-color: {{ .BackgroundColor }};
+						color: {{ .ForegroundColor }};
+					}
+				</style>
+			</head>
+			
+			<body>
+				<div id="root">
+					<h1>{{ .Title }}</h1>
+					<a href="/?query={{ .Query }}">Search for "{{ .Query }}"</a>
+					<ul>
+						<li>{{ index .State.todos 0 }}</li>
+						<li>{{ index .State.todos 1 }}</li>
+					</ul>
+				</div>
+				<script src="/app.js"></script>
+				<script>
+					const state = {{ .State }};
+					renderApp(state);
+				</script>
+			</body>
+			
+			</html>
+			`
+			expectation := `
+			<!DOCTYPE html>
+			<html lang="en">
+			
+			<head>
+				<meta charset="utf-8" />
+				<title>My App</title>
+				<style type="text/css">
+					body {
+						background-color: ZgotmplZ;
+						color: #e0e0ff;
+					}
+				</style>
+			</head>
+			
+			<body>
+				<div id="root">
+					<h1>Welcome to &lt;b&gt;My App&lt;/b&gt;.</h1>
+					<a href="/?query=Lazy%20dog%20%3e%20Quick%20brown%20fox%3f">Search for "Lazy dog &gt; Quick brown fox?"</a>
+					<ul>
+						<li>&lt;b&gt;Important things! \o/&lt;/b&gt;</li>
+						<li>Cats &amp; Dogs</li>
+					</ul>
+				</div>
+				<script src="/app.js"></script>
+				<script>
+					const state = {"query":"Lazy dog \u003e Quick brown fox?","todos":["\u003cb\u003eImportant things! \\o/\u003c/b\u003e","Cats \u0026 Dogs"]};
+					renderApp(state);
+				</script>
+			</body>
+			
+			</html>
+			`
+
+			out, err := ParseHTMLTemplate("test", template, map[string]interface{}{
+				"URL":             "https://www.example.com",
+				"Title":           "Welcome to <b>My App</b>.",
+				"BackgroundColor": "#101020; /* for contrast */",
```

Comment:
Why is BackgroundColor considered unsafe?

Reply:
The template placeholder is in CSS value context, so only plain CSS value is allowed: comment is disallowed.